src/components/appointments/TimeSelect.js

```diff
@@ -20,14 +21,14 @@ const getDatetimes = async (baseUrl, productIds, locationId, date) => {
   return datetimesList.map(item => item.time);
 };
 
-const TimeSelect = () => {
+const TimeSelect = ({products}) => {
   const {baseUrl} = useContext(ConfigContext);
   const {values} = useFormikContext();
   const calendarLocale = useCalendarLocale();
 
   const getOptions = useCallback(
     async () => {
-      const productIds = (values.products || []).map(prod => prod.productId);
+      const productIds = products.map(prod => prod.productId);
```

Comment:
Should we splitup the stringify in the dependency list of this callback to somehting like
```js
    [baseUrl, calendarLocale, JSON.stringify(values.location), values.date]
```
or whatever makes sense for the shape of these...

I'm a bit concerned this pattern of depending on the complete form state will get copied to bigger form states.

Reply:
I had a ticket #453 where I intended to clean those up yeah!